controllers/front/ProductController.php

```diff
@@ -985,6 +990,41 @@ function ($elem) use ($checkedIdProductAttribute) {
             );
 
             if (empty($availableProductAttribute) && count($availableProductAttributes)) {
+                if (!Product::isAvailableWhenOutOfStock($this->product->out_of_stock)) {
```

Comment:
It would be great to add some comments to explain what is done here

Reply:
comments added @sowbiba